pkg/api/types.go

```diff
@@ -2700,6 +2707,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// Optional: Represents the configurations of Client IP based session affinity.
```

Comment:
I think you want a struct, for future safety.

```
type ServiceAffinityConfig struct {
  // only one of the following may be set
  ClientIPConfig *ServiceAffinityConfigClientIP
}

type ServiceAffinityConfigClientIP struct {
  TimeoutSeconds int
}
```

Then embed ServiceAffinityConfig